files/scripts/supervisor-proc-exit-listener

```diff
@@ -49,6 +54,106 @@ def get_critical_group_and_process_list():
     return critical_group_list, critical_process_list
 
 
+def get_command_result(command):
+    """
+    @summary: Execute the command and return result.
+    @return: A string which contains the execution result.
+    """
+    command_stdout = ""
+
+    try:
+        proc_instance = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                         shell=True, universal_newlines=True)
+        command_stdout, command_stderr = proc_instance.communicate()
+        if proc_instance.returncode != 0:
+            syslog.syslog(syslog.LOG_ERR, "Failed to execute the command '{}'. Return code: '{}'".format(
+                command, proc_instance.returncode))
+            sys.exit(7)
+    except OSError as os_err:
+        syslog.syslog(syslog.LOG_ERR, "Failed to execute the command '{}'. Error: '{}'".format(command, os_err))
+        sys.exit(8)
+    except ValueError as val_err:
+        syslog.syslog(syslog.LOG_ERR, "Failed to execute the command '{}'. Error: '{}'".format(command, val_err))
+        sys.exit(9)
+
+    return command_stdout
+
+
+def is_process_running(process_name):
+    """
+    @summary: Determine whether a critical process was running or not.
+    @return: Return 'True' if process was runnning. Otherwise return 'False'.
+    """
+    supervisorctl_status_command = "supervisorctl status"
+    command_stdout = ""
+    is_running = False
+
+    command_stdout = get_command_result(supervisorctl_status_command)
+    for line in command_stdout.split("\n"):
+        if process_name in line:
+            status = line.split()[1].strip()
+            if status == "RUNNING":
+                is_running = True
+            break
+
+    return is_running
+
+
+def generate_alerting_message(process_name):
+    """
+    @summary: If a critical process was not running, this function will determine it was running in host
+              or in a specific namespace. Then an alerting message will be written into syslog.
+    """
+    namespace_prefix = os.environ.get("NAMESPACE_PREFIX")
+    namespace_id = os.environ.get("NAMESPACE_ID")
+
+    if not namespace_prefix or not namespace_id:
+        syslog.syslog(syslog.LOG_ERR, "Process '{}' is not running in 'host'.".format(process_name))
+    else:
+        namespace = namespace_prefix + namespace_id
+        syslog.syslog(
+            syslog.LOG_ERR, "Process '{}' is not running in namespace '{}'.".format(process_name, namespace))
+
+
+def monitoring_process_and_alerting(process_name):
+    """
+    @summary: This function  will determine whether a critical process was running or not in every minute.
+              If it was not running, then an alerting message will be writen into syslog. Otherwise,
+              this function will exit.
+    """
+    while True:
+        time.sleep(60)
+        if not is_process_running(process_name):
```

Comment:
if this process is not running, then you are trapped in this loop, you never get it out.

Reply:
If a critical process was not running, event listener need periodically (every 1 minute) write an alerting messages into syslog like Monit, right?